internal/parser.go

```diff
@@ -902,8 +902,10 @@ func inHeadIM(p *parser) bool {
 			p.im = afterHeadIM
 			return true
 		case a.Body, a.Html, a.Br:
-			p.parseImpliedToken(EndTagToken, a.Head, a.Head.String())
```

Comment:
The fix was to remove this divergence from the [html spec](https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-inhead:~:text=tag%20name%20is%20one%20of%3A%20%22body%22%2C%20%22html%22%2C%20%22br%22-,Act%20as%20described%20in%20the%20%22anything%20else%22%20entry%20below.,-A%20start%20tag%20whose%20tag%20name%20is%20%22template) when in the `inHead` insertion mode, see the ["any thing"](https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-inhead:~:text=Ignore%20the%20token.-,Anything%20else,-Pop%20the%20current) section to find the correct steps to be performed